diff -ur LVM2.2.02.136_O/tools/Makefile.in LVM2.2.02.136/tools/Makefile.in
--- LVM2.2.02.136_O/tools/Makefile.in	2015-12-16 19:50:03.205143738 +0000
+++ LVM2.2.02.136/tools/Makefile.in	2015-12-16 19:51:06.115782426 +0000
@@ -135,7 +135,7 @@
 
 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L$(interfacebuilddir) \
-	      -o $@ dmsetup.o -ldevmapper $(STATIC_LIBS) $(LIBS)
+	      -o $@ dmsetup.o -ldevmapper -lm $(STATIC_LIBS) $(LIBS)
 
 all: device-mapper
 
