--- configure.in	2011-06-21 09:27:34.000000000 +0300
+++ configure.in	2011-11-30 14:48:27.000000000 +0200
@@ -65,16 +65,6 @@
 dnl Checks for libraries.
 AC_CHECK_LIB(pthread, pthread_create)
 
-PKG_CHECK_MODULES(GTK, gtk+-2.0,
-   [
-    AC_DEFINE([HAVE_GTK_2], 1,
-	      [Define to 1 if you have the gtk+-2.0 library.])
-   ],[
-    PKG_CHECK_MODULES(GTK, gtk+) 
-    PKG_CHECK_MODULES(GDK_IMLIB, imlibgdk)
-   ])
-
-
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_HEADER_DIRENT
@@ -132,8 +122,6 @@
 dnl  appendix
 
 AC_DEFINE(CONV_VERSION2,1,CONV_VERSION2)
-AC_DEFINE(USE_GTK,HAVE_LIBGTK,
-	  [FIXME: Alias for HAVE_LIBGTK for as long as we need it])
 AC_DEFINE(_V1_SUPPORT,1,_V1_SUPPORT)
 AC_DEFINE(DEBUG,0,DEBUG)
 AC_DEFINE(ROLL_PAPER,1,ROLL_PAPER)
@@ -179,7 +167,6 @@
 src/filter-l2/Makefile \
 daemon/Makefile \
 daemon/rc/Makefile \
-status-monitor/Makefile \
 layout_script/Makefile \
 pixmaps/Makefile \
 ppd/Makefile \

--- Makefile.am	2010-09-02 08:12:00.000000000 +0300
+++ Makefile.am	2011-11-30 15:23:45.000000000 +0200
@@ -16,7 +16,7 @@
 ## along with this program; if not, write to the Free Software
 ## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-SUBDIRS = lib lib-l2 src daemon status-monitor \
+SUBDIRS = lib lib-l2 src daemon \
           layout_script po ppd \
           setup pixmaps makeinstall

--- src/Makefile.am	2010-09-02 08:12:00.000000000 +0300
+++ src/Makefile.am	2011-11-30 15:06:48.000000000 +0200
@@ -45,7 +45,6 @@
 # CUPS
 AM_CFLAGS += -DCUPS_FILTER_PATH=\"$(cupsfilterdir)\"
 
-INCLUDES  = @GTK_CFLAGS@ 
-INCLUDES += -I$(top_srcdir)/lib
+INCLUDES = -I$(top_srcdir)/lib
 libcutils_la_SOURCES = \
 	gettext.h \

--- makeinstall/Makefile.am	2010-09-02 08:12:00.000000000 +0300
+++ makeinstall/Makefile.am	2011-11-30 15:33:27.000000000 +0200
@@ -19,22 +19,17 @@
 pkgscriptdir = $(pkglibdir)/scripts
 pkgcachedir = $(localstatedir)/cache/$(PACKAGE_TARNAME)
 
-pkglib_PROGRAMS = pipslite-install
-
 AM_CFLAGS = -DPRTOPT_PATH=\"$(pkgcachedir)/prtOpt.csv\"
 AM_CFLAGS += -DPAPER_PATH=\"$(pkgdatadir)/paper_list.csv\"
 AM_CFLAGS += -DCACHEDIR_PATH=\"$(pkgcachedir)/\"
 AM_CFLAGS += -DPPDINSTALL_SH=\"$(pkgscriptdir)/install-ppd.sh\"
 
 INCLUDES = \
-	-I$(top_srcdir)/src \
-	-I$(top_srcdir)/status-monitor \
-	@GTK_CFLAGS@
+	-I$(top_srcdir)/src
 
 pipslite_install_SOURCES = \
 	files_parser.c \
 	ppd_generator.c \
-	ppd_tool.c \
 	../src/prtOpt.h \
 	utility.c \
 	../src/paper.rsc \
@@ -42,17 +37,14 @@
 	../src/paper.h \
 	rscCtl.c \
 	pipsError.c \
-	support.c \
 	definitions.h \
 	files_parser.h \
 	ppd_generator.h \
 	utility.h \
 	rscCtl.h \
-	pipsError.h \
-	support.h
+	pipsError.h
 
 pipslite_install_LDADD = \
-	@GTK_LIBS@ \
 	$(INTLLIBS) \
 	$(LIBLTDL) \
 	$(top_builddir)/src/getstat.o \
