diff -Naurd sixfireusb-0.6.2/pcm.c sixfireusb-0.6.2_5.7/pcm.c
--- a/pcm.c	2013-02-04 03:17:43.000000000 +0100
+++ b/pcm.c	2019-06-03 08:43:50.668438334 +0200
@@ -202,7 +202,6 @@
 {
 	struct pcm_runtime *rt = kzalloc(sizeof(struct pcm_runtime), GFP_KERNEL);
 	struct substream_runtime *sub_rt = chip->substream;
-	int ret;
 	int i;
 	
 	if (!rt)
@@ -216,16 +215,10 @@
 		sub_rt->devices[i]->private_data = chip;
 		snd_pcm_set_ops(sub_rt->devices[i], SNDRV_PCM_STREAM_PLAYBACK, &pcm_ops);
 		snd_pcm_set_ops(sub_rt->devices[i], SNDRV_PCM_STREAM_CAPTURE, &pcm_ops);
-		ret = snd_pcm_lib_preallocate_pages_for_all(sub_rt->devices[i],
+		snd_pcm_lib_preallocate_pages_for_all(sub_rt->devices[i],
 				SNDRV_DMA_TYPE_CONTINUOUS,
 				snd_dma_continuous_data(GFP_KERNEL),
 				MAX_BUFSIZE, MAX_BUFSIZE);
-		if (ret) {
-			kfree(rt);
-			snd_printk(KERN_ERR PREFIX
-					"error preallocating pcm buffers.\n");
-			return ret;
-		}
 	}
 
 	chip->pcm = rt;
--- a/control.c	2013-02-04 04:18:31.000000000 +0200
+++ b/control.c	2021-01-17 18:05:10.403381568 +0200
@@ -571,7 +571,7 @@
 		ret = snd_ctl_add(card, control);
 		if (ret < 0)
 			return ret;
-		ret = snd_ctl_add_slave(vmaster, control);
+		ret = snd_ctl_add_follower(vmaster, control);
 		if (ret < 0)
 			return ret;
 		i++;
--- a/chip.c	2015-11-28 14:59:01.000000000 +0200
+++ b/chip.c	2021-10-20 20:15:45.893538454 +0300
@@ -33,7 +33,6 @@
 MODULE_AUTHOR("Torsten Schenk <torsten.schenk@zoho.com>");
 MODULE_DESCRIPTION("TerraTec DMX 6Fire USB audio driver, version 0.6.1");
 MODULE_LICENSE("GPL v2");
-MODULE_SUPPORTED_DEVICE("{{TerraTec, DMX 6Fire USB}}");
 
 static int index[SNDRV_CARDS] = SNDRV_DEFAULT_IDX; /* Index 0-max */
 static char *id[SNDRV_CARDS] = SNDRV_DEFAULT_STR; /* Id for card */
