--- dockapp_imlib2.c	2004-07-15 20:51:08.000000000 +0300
+++ dockapp_imlib2.c.new	2006-09-05 22:50:54.000000000 +0300
@@ -246,6 +246,7 @@
   /* some default font paths */
   snprintf(fp, 512, "%s/.fonts", getenv("HOME"));
   add_fontpath(fp,1);
+  add_fontpath("/usr/share/fonts/corefonts",1);
   add_fontpath("/usr/share/fonts/truetype",1);
   add_fontpath("/usr/share/fonts/ttf",1);
   add_fontpath("/usr/X11R6/lib/X11/fonts/truetype",1);
