--- a/SConstruct	2019-06-19 16:48:10.000000000 +0300
+++ b/SConstruct	2019-09-21 17:25:50.945132461 +0300
@@ -16,8 +16,18 @@
 env.Replace(LIBS="")
 env.Replace(LIBPATH="")
 
-env.MergeFlags(['-Wall', '-Werror', '-g', '-fpic', '-std=c++11'])
-
+# ---- check for environment variables
+if 'CXX' in os.environ:
+	env.Replace(CXX = os.environ['CXX'])
+	print(">> Using compiler " + os.environ['CXX'])
+
+if 'CXXFLAGS' in os.environ:
+	env.Append(CCFLAGS = os.environ['CXXFLAGS'])
+	print(">> Appending custom build flags : " + os.environ['CXXFLAGS'])
+
+if 'LDFLAGS' in os.environ:
+	env.Append(LINKFLAGS = os.environ['LDFLAGS'])
+	print(">> Appending custom link flags : " + os.environ['LDFLAGS'])
 tests = {}
 tests.update(env['PKGCONFIG_TESTS'])
 
