--- a/setup.py	2015-10-31 16:22:49.000000000 +0200
+++ b/setup.py	2018-11-07 12:35:50.896048341 +0200
@@ -187,6 +187,8 @@
 
             if pkg_config.wait() == 0:
                 try:
+                    if "+" in pkg_config_stdout:  # cdparanoia new versions
+                        return tuple(int(s) for s in pkg_config_stdout.split("+")[1].split("."))
                     return tuple(int(s) for s in pkg_config_stdout.split("."))
                 except ValueError:
                     # version isn't all integers
@@ -481,33 +481,19 @@
         libraries = set()
         extra_compile_args = []
         extra_link_args = []
+        paranoia_version = []
 
         if system_libraries.present("libcdio_paranoia"):
             if system_libraries.guaranteed_present("libcdio_paranoia"):
                 libraries.update(set(["libcdio",
                                       "libcdio_cdda",
                                       "libcdio_paranoia"]))
-                try:
-                    if tuple(int(s) for s in
-                             system_libraries.configfile.get(
-                                 "Libraries",
-                                 "libcdio_paranoia_version")) < (0, 90):
-                        paranoia_version = [("OLD_PARANOIA", None)]
-                    else:
-                        paranoia_version = []
-                except (KeyError, ValueError):
-                    paranoia_version = []
             else:
                 extra_compile_args.extend(
                     system_libraries.extra_compile_args("libcdio_paranoia"))
                 extra_link_args.extend(
                     system_libraries.extra_link_args("libcdio_paranoia"))
 
-                if system_libraries.lib_version("libcdio_paranoia") < (0, 90):
-                    paranoia_version = [("OLD_PARANOIA", None)]
-                else:
-                    paranoia_version = []
-
             sources.extend(["src/cdiomodule.c",
                             "src/framelist.c",
                             "src/pcm_conv.c"])
@@ -519,7 +505,6 @@
             self.__library_manifest__.append(("libcdio",
                                               "CDDA data extraction",
                                               False))
-            paranoia_version = []
 
         Extension.__init__(
             self,
