--- libedcast/libedcast.cpp	2008-05-02 06:08:02.000000000 +0300
+++ libedcast/libedcast.cpp.new	2009-02-04 00:52:28.000000000 +0200
@@ -2402,7 +2402,7 @@
 
 		FLAC__stream_encoder_set_channels(g->flacEncoder, g->currentChannels);
 
-		
+		FLAC__stream_encoder_set_sample_rate(g->flacEncoder, g->currentSamplerate);
 /*
 		FLAC__stream_encoder_set_write_callback(g->flacEncoder,(FLAC__StreamEncoderWriteCallback) FLACWriteCallback,
 												   (FLAC__StreamEncoderWriteCallback) FLACWriteCallback);
