From aad18cc7e6bb71d2f7c9ada3b427cbcef0afc9e0 Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Thu, 9 Jan 2014 11:29:27 +0100
Subject: [PATCH] decoder: switch to the avcodec_decode_audio4() API

---
 decoder.h | 43 ++++++++++++++++++++++++++++---------------
 1 file changed, 28 insertions(+), 15 deletions(-)

diff --git a/decoder.h b/decoder.h
index effb2bf..e61602e 100644
--- a/decoder.h
+++ b/decoder.h
@@ -28,6 +28,12 @@
 extern "C" {
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
+
+#define NEW_AVFRAME_API (LIBAVCODEC_VERSION_MAJOR >= 55)
+#if NEW_AVFRAME_API
+#include <libavutil/frame.h>
+#endif
+
 #ifdef HAVE_AV_AUDIO_CONVERT
 #include "ffmpeg/audioconvert.h"
 #include "ffmpeg/samplefmt.h"
@@ -64,7 +70,6 @@ public:
 
 private:
 	static const int BUFFER_SIZE = AVCODEC_MAX_AUDIO_FRAME_SIZE * 2;
-	uint8_t *m_buffer1;
 	uint8_t *m_buffer2;
 	std::string m_file_name;
 	std::string m_error;
@@ -73,6 +78,7 @@ private:
 	bool m_codec_open;
 	AVStream *m_stream;
     static QMutex m_mutex;
+    AVFrame *m_frame;
 #ifdef HAVE_AV_AUDIO_CONVERT
 	AVAudioConvert *m_convert_ctx;
 #endif
@@ -110,8 +116,13 @@ inline Decoder::Decoder(const std::string &file_name)
 	, m_convert_ctx(0)
 #endif
 {
-	m_buffer1 = (uint8_t *)av_malloc(BUFFER_SIZE + 16);
 	m_buffer2 = (uint8_t *)av_malloc(BUFFER_SIZE + 16);
+
+#if NEW_AVFRAME_API
+    m_frame = av_frame_alloc();
+#else
+    m_frame = avcodec_alloc_frame();
+#endif
 }
 
 inline Decoder::~Decoder()
@@ -128,8 +139,13 @@ inline Decoder::~Decoder()
 		av_audio_convert_free(m_convert_ctx);
 	}
 #endif
-	av_free(m_buffer1);
 	av_free(m_buffer2);
+
+#if NEW_AVFRAME_API
+    av_frame_free(&m_frame);
+#else
+    av_freep(&m_frame);
+#endif
 }
 
 inline bool Decoder::Open()
@@ -220,11 +236,9 @@ inline void Decoder::Decode(FingerprintCalculator *consumer, int max_length)
 		packet_temp.data = packet.data;
 		packet_temp.size = packet.size;
 		while (packet_temp.size > 0) {
-			int buffer_size = BUFFER_SIZE;
-                        int consumed = avcodec_decode_audio3(
-                                m_codec_ctx, (int16_t *)m_buffer1, &buffer_size,
-                                &packet_temp);
-
+            int got_output;
+            int consumed = avcodec_decode_audio4(m_codec_ctx, m_frame,
+                                                 &got_output, &packet_temp);
 			if (consumed < 0) {
 				break;
 			}
@@ -232,32 +246,31 @@ inline void Decoder::Decode(FingerprintCalculator *consumer, int max_length)
 			packet_temp.data += consumed;
 			packet_temp.size -= consumed;
 
-			if (buffer_size <= 0) {
+			if (!got_output) {
 				continue;
 			}
 
 			int16_t *audio_buffer;
 #ifdef HAVE_AV_AUDIO_CONVERT
 			if (m_convert_ctx) {
-				const void *ibuf[6] = { m_buffer1 };
+				const void *ibuf[6] = { m_frame->data[0] };
 				void *obuf[6] = { m_buffer2 };
 				int istride[6] = { av_get_bytes_per_sample(m_codec_ctx->sample_fmt) };
 				int ostride[6] = { 2 };
-				int len = buffer_size / istride[0];
+				int len = m_frame->nb_samples;
 				if (av_audio_convert(m_convert_ctx, obuf, ostride, ibuf, istride, len) < 0) {
 					break;
 				}
 				audio_buffer = (int16_t *)m_buffer2;
-				buffer_size = len * ostride[0];
 			}
 			else {
-				audio_buffer = (int16_t *)m_buffer1;
+				audio_buffer = (int16_t *)m_frame->data[0];
 			}
 #else
-			audio_buffer = (int16_t *)m_buffer1;
+			audio_buffer = (int16_t *)m_frame->data[0];
 #endif
 
-			int length = buffer_size / 2;
+			int length = m_frame->nb_samples;
 			if (max_length) {
 				length = std::min(remaining, length);
 			}
-- 
1.8.5.2

