From 1c1bf578963745a95818db11374b62f3c185cf35 Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Thu, 9 Jan 2014 10:40:00 +0100
Subject: [PATCH] decoder: drop support for libav* libraries before libav 0.8

---
 decoder.h | 34 ----------------------------------
 1 file changed, 34 deletions(-)

diff --git a/decoder.h b/decoder.h
index 028f58f..effb2bf 100644
--- a/decoder.h
+++ b/decoder.h
@@ -35,10 +35,6 @@ extern "C" {
 }
 #include "fingerprintcalculator.h"
 
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(52, 64, 0)
-#define AV_SAMPLE_FMT_S16 SAMPLE_FMT_S16
-#endif
-
 class Decoder
 {
 public:
@@ -125,11 +121,7 @@ inline Decoder::~Decoder()
 		avcodec_close(m_codec_ctx);
 	}
 	if (m_format_ctx) {
-#if LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(53, 17, 0)
-		av_close_input_file(m_format_ctx);
-#else
 		avformat_close_input(&m_format_ctx);
-#endif
 	}
 #ifdef HAVE_AV_AUDIO_CONVERT
 	if (m_convert_ctx) {
@@ -144,20 +136,12 @@ inline bool Decoder::Open()
 {
     QMutexLocker locker(&m_mutex); 
 
-#if LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(53, 2, 0)
-	if (av_open_input_file(&m_format_ctx, m_file_name.c_str(), NULL, 0, NULL) != 0) {
-#else
 	if (avformat_open_input(&m_format_ctx, m_file_name.c_str(), NULL, NULL) != 0) {
-#endif
 		m_error = "Couldn't open the file." + m_file_name;
 		return false;
 	}
 
-#if LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(53, 6, 0)
-	if (av_find_stream_info(m_format_ctx) < 0) {
-#else
 	if (avformat_find_stream_info(m_format_ctx, NULL) < 0) {
-#endif
 		m_error = "Couldn't find stream information in the file.";
 		return false;
 	}
@@ -166,11 +150,7 @@ inline bool Decoder::Open()
 
 	for (int i = 0; i < m_format_ctx->nb_streams; i++) {
 		AVCodecContext *avctx = m_format_ctx->streams[i]->codec;
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(52, 64, 0)
-                if (avctx && avctx->codec_type == CODEC_TYPE_AUDIO) {
-#else
                 if (avctx && avctx->codec_type == AVMEDIA_TYPE_AUDIO) {
-#endif
 			m_stream = m_format_ctx->streams[i];
 			m_codec_ctx = avctx;
 			break;
@@ -187,11 +167,7 @@ inline bool Decoder::Open()
 		return false;
 	}
 
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(53, 8, 0)
-    if (avcodec_open(m_codec_ctx, codec) < 0) {
-#else
 	if (avcodec_open2(m_codec_ctx, codec, NULL) < 0) {
-#endif
         m_error = "Couldn't open the codec.";
         return false;
     }
@@ -245,15 +221,9 @@ inline void Decoder::Decode(FingerprintCalculator *consumer, int max_length)
 		packet_temp.size = packet.size;
 		while (packet_temp.size > 0) {
 			int buffer_size = BUFFER_SIZE;
-#if LIBAVCODEC_VERSION_INT <= AV_VERSION_INT(52, 25, 0)
-                        int consumed = avcodec_decode_audio2(
-                                m_codec_ctx, (int16_t *)m_buffer1, &buffer_size,
-                                packet_temp.data, packet_temp.size);
-#else
                         int consumed = avcodec_decode_audio3(
                                 m_codec_ctx, (int16_t *)m_buffer1, &buffer_size,
                                 &packet_temp);
-#endif
 
 			if (consumed < 0) {
 				break;
@@ -271,11 +241,7 @@ inline void Decoder::Decode(FingerprintCalculator *consumer, int max_length)
 			if (m_convert_ctx) {
 				const void *ibuf[6] = { m_buffer1 };
 				void *obuf[6] = { m_buffer2 };
-#if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(51, 8, 0)
-				int istride[6] = { av_get_bits_per_sample_format(m_codec_ctx->sample_fmt) / 8 };
-#else
 				int istride[6] = { av_get_bytes_per_sample(m_codec_ctx->sample_fmt) };
-#endif
 				int ostride[6] = { 2 };
 				int len = buffer_size / istride[0];
 				if (av_audio_convert(m_convert_ctx, obuf, ostride, ibuf, istride, len) < 0) {
-- 
1.8.5.2

