--- ./SConstruct.ori	2015-08-01 03:47:54.506016655 +0200
+++ ./SConstruct	2015-08-01 03:42:10.003017365 +0200
@@ -7,15 +7,15 @@
 ###############################################
 # options
 ipager_optfile = [ 'scons.opts', 'user.opts' ]
-ipager_options = Options(ipager_optfile)
-ipager_options.AddOptions(
-    BoolOption('debug', 'build debug version', 0),
-    BoolOption('debug_events', 'debug xserve events', 0),
+ipager_options = Variables(ipager_optfile)
+ipager_options.AddVariables(
+    BoolVariable('debug', 'build debug version', 0),
+    BoolVariable('debug_events', 'debug xserve events', 0),
 
-    BoolOption('xinerama', 'support xinerama', 0),
+    BoolVariable('xinerama', 'support xinerama', 0),
 
-    PathOption('PREFIX', 'install-path base', '/usr/local'),
-    PathOption('DESTDIR', 'install to $DESTDIR/$PREFIX', '/')
+    PathVariable('PREFIX', 'install-path base', '/usr/local'),
+    PathVariable('DESTDIR', 'install to $DESTDIR/$PREFIX', '/')
 )
 
 
