--- rr_lib.rb	2010-09-19 10:08:13.000000000 +0300
+++ rr_lib.rb	2010-09-20 07:39:13.000000000 +0300
@@ -380,25 +380,25 @@
 		if matches_all != matches_errors; addLog(_("Chunks that differed after %s trials,\nwere tried to match %s times.\n") % [matches_all, matches_errors], true) end
 
 		if @problem_tracks.empty?
- 			addLog(_("None of the tracks gave any problems\n"), true)
- 		elsif @not_corrected_tracks.size != 0
- 			addLog(_("Some track(s) could NOT be corrected within the maximum amount of trials\n"), true)
- 			@not_corrected_tracks.each do |track|
+			addLog(_("None of the tracks gave any problems\n"), true)
+		elsif @not_corrected_tracks.size != 0
+			addLog(_("Some track(s) could NOT be corrected within the maximum amount of trials\n"), true)
+			@not_corrected_tracks.each do |track|
 				@rippingErrors = true
 				addLog(_("Track %s could NOT be corrected completely\n") % [track], true)
 			end
- 		else
- 			addLog(_("Some track(s) needed correction,but could\nbe corrected within the maximum amount of trials\n"), true)
- 		end
+		else
+			addLog(_("Some track(s) needed correction,but could\nbe corrected within the maximum amount of trials\n"), true)
+		end
 
- 		if !@problem_tracks.empty? # At least some correction was necessary
+		if !@problem_tracks.empty? # At least some correction was necessary
 			position_analyse(matches_errors, maxtries)
 			@short_summary += _("The exact positions of the suspicious chunks\ncan be found in the ripping log\n")
 		end
 		@logfiles.each{|logfile| logfile.close} #close all the files
- 	end
- 		
- 	def position_analyse(matches_errors, maxtries) # Give an overview of suspicion position in the logfile
+	end
+
+	def position_analyse(matches_errors, maxtries) # Give an overview of suspicion position in the logfile
 		addLog(_("\nSUSPICIOUS POSITION ANALYSIS\n\n"))
 		addLog(_("Since there are 75 chunks per second, after making the notion of the\n"))
 		addLog(_("suspicious position, the amount of initially mismatched chunks for\nthat position is shown.\n\n"))
@@ -645,12 +645,12 @@
 	end
 
 	def getFreedbString
-		if installed('discid')
+		if installed('cd-discid') # try this first as it works with freedb.musicbrainz.org
+			@freedbString = `cd-discid #{@cdrom}`
+		elsif installed('discid') # if no cd-discid exists, try discid
 			if RUBY_PLATFORM.include?('darwin') ; `diskutil unmount #{@cdrom}` end
 			@freedbString = `discid #{@cdrom}`
 			if RUBY_PLATFORM.include?('darwin') ; `diskutil mount #{@cdrom}` end
-		elsif installed('cd-discid') # if no discid exists, try cd-discid
-			@freedbString = `cd-discid #{@cdrom}`
 		else # if both do not exist generate it ourselves (less foolproof)
 			puts _("warning: discid or cd-discid isn't found on your system! Using fallback...")
 			checkDataTrack()
@@ -1848,12 +1848,12 @@
 
 	def sizeTest(track)
 		puts "Expected filesize for #{if track == "image" ; track else "track #{track}" end}\
-		is #{@settings['cd'].getFileSize(track)} bytes." if @settings['debug'] 
+		is #{@settings['cd'].getFileSize(track)} bytes." if @settings['debug']
 
-		if installed('df')				
-			freeDiskSpace = `LANG=C df \"#{@settings['Out'].getDir()}\"`.split()[10].to_i
-			puts "Free disk space is #{freeDiskSpace} MB" if @settings['debug']
-			if @settings['cd'].getFileSize(track) > freeDiskSpace*1000
+		if installed('df')
+			freeDiskSpace = `LANG=C df -Pk \"#{@settings['Out'].getDir()}\"`.split()[9].to_i * 1024
+			puts "Free disk space is #{freeDiskSpace / 1000 / 1000} MB" if @settings['debug']
+			if @settings['cd'].getFileSize(track) > freeDiskSpace
 				@settings['log'].add(_("Not enough disk space left! Rip aborted"))
 				return false
 			end
