--- rubyripper_cli.rb	2010-09-19 10:08:13.000000000 +0300
+++ rubyripper_cli.rb	2011-05-16 19:56:12.000000000 +0300
@@ -19,13 +19,15 @@
 
 found_rrlib = false
 RUBYDIR.each do |dir|
-  if File.exist?(file = File.join(dir, '/rr_lib.rb'))
-	require file; found_rrlib = true ; break
-  end
+	if File.exist?(file = File.join(dir, '/rr_lib.rb'))
+		require file; found_rrlib = true ; break
+	end
 end
+
+
 if found_rrlib == false
-  puts "The main program logic file 'rr_lib.rb' can't be found!"
-  exit()
+	puts "The main program logic file 'rr_lib.rb' can't be found!"
+	exit()
 end
 
 require 'optparse'
@@ -100,18 +102,18 @@
 		puts _("*** RUBYRIPPER SETTINGS ***")
 		puts ""
 		puts _("** ENCODING **")
- 		puts _("1)  flac [%s] , settings: %s") % [@settings['flac'] ? '*' : ' ', @settings['flacsettings']]
- 		puts _("2)  vorbis [%s] , settings: %s") % [@settings['vorbis'] ? '*' : ' ', @settings['vorbissettings']]
- 		puts _("3)  mp3 [%s] , settings: %s") % [@settings['mp3'] ? '*' : ' ', @settings['mp3settings']]
- 		puts _("4)  wav [%s]") % [@settings['wav'] ? '*' : ' ']
- 		puts _("5)  other [%s], settings: %s") % [@settings['othersettings'], @settings['other'] ? '*' : ' '] 
- 		puts _("6)  playlist support [%s]") % [@settings['playlist'] ? '*' : ' ']
- 		puts _("7)  encode while ripping [%s]")  % [@settings['maxThreads']]
- 		puts ""
- 		puts _("** RIPPING **")
- 		puts _("8)  cdrom : %s with offset %s") % [@settings['cdrom'], @settings['offset']]
+		puts _("1)  flac [%s] , settings: %s") % [@settings['flac'] ? '*' : ' ', @settings['flacsettings']]
+		puts _("2)  vorbis [%s] , settings: %s") % [@settings['vorbis'] ? '*' : ' ', @settings['vorbissettings']]
+		puts _("3)  mp3 [%s] , settings: %s") % [@settings['mp3'] ? '*' : ' ', @settings['mp3settings']]
+		puts _("4)  wav [%s]") % [@settings['wav'] ? '*' : ' ']
+		puts _("5)  other [%s], settings: %s") % [@settings['othersettings'], @settings['other'] ? '*' : ' '] 
+		puts _("6)  playlist support [%s]") % [@settings['playlist'] ? '*' : ' ']
+		puts _("7)  encode while ripping [%s]")  % [@settings['maxThreads']]
+		puts ""
+		puts _("** RIPPING **")
+		puts _("8)  cdrom : %s with offset %s") % [@settings['cdrom'], @settings['offset']]
 		puts _("   **Find your offset at http://www.accuraterip.com/driveoffsets.htm.\n   **Your drive model is shown in the logfile.")
- 		puts _("9)  passing extra cdparanoia parameters : %s") % [@settings['rippersettings'] != '' ? @settings['rippersettings'] : 'none']
+		puts _("9)  passing extra cdparanoia parameters : %s") % [@settings['rippersettings'] != '' ? @settings['rippersettings'] : 'none']
 		puts _("10) match all chunks : %s") % [@settings['req_matches_all']]
 		puts _("11) match erroneous chunks : %s") % [@settings['req_matches_errors']]
 		puts _("12) maximum trials : %s") % [@settings['max_tries'] == 0 ? "no maximum" : @settings['max_tries']]
@@ -121,17 +123,17 @@
 		puts _("14) base directory : %s") % [@settings['basedir']]
 		puts _("15) standard scheme: %s") % [@settings['naming_normal']]
 		puts _("16) various artist scheme: %s") % [@settings['naming_various']]
- 		puts ""
- 		puts _("** FREEDB  **")
+		puts ""
+		puts _("** FREEDB  **")
 		puts _("17) fetch cd info with freedb [%s]") % [@settings['freedb'] ? '*' : ' ']
 		puts _("18) always use first hit [%s]") % [@settings['first_hit'] ? '*' : ' ']
 		puts _("19) site = %s") % [@settings['site']]
 		puts _("20) username = %s") % [@settings['username']]
 		puts _("21) hostname = %s") % [@settings['hostname']]
- 		puts ""
+		puts ""
 		puts _("88) config file = %s") % [@options.file]
- 		puts _("99) Don't change any setting")
- 		puts ""
+		puts _("99) Don't change any setting")
+		puts ""
 	end
 
 	def edit_settings # depending on the input of the user as asked in show_settings(), change the setting
@@ -191,18 +193,18 @@
 		if variable == 'normal'
 			puts _("\nCurrent naming scheme: %s") % [@settings['naming_normal']]
 			puts get_example_filename_normal(@settings['basedir'], @settings['naming_normal'])
-		else 
+		else
 			puts _("\nCurrent naming scheme: %s") % [@settings['naming_various']]
 			puts get_example_filename_various(@settings['basedir'], @settings['naming_various'])
 		end
 
-		puts	_("\n%a = Artist	\n%b = Album\n%g = Genre\n%y = Year\n%f = Codec\n%n = Tracknumber\n%t = Trackname\n%va = Various Artist\n")
-		answer = get_answer(_("New %s naming scheme (q to quit) : ") % [variable], "open", "%f/%a (%y) %b/%n - %t") 
+		puts _("\n%a = Artist\n%b = Album\n%g = Genre\n%y = Year\n%f = Codec\n%n = Tracknumber\n%t = Trackname\n%va = Various Artist\n")
+		answer = get_answer(_("New %s naming scheme (q to quit) : ") % [variable], "open", "%f/%a (%y) %b/%n - %t")
 		if answer !=_('q')
 			if variable == 'normal'
 				puts _("An example filename is now:\n\n\t%s") % [get_example_filename_normal(@settings['basedir'], answer)]
 				@settings['naming_normal'] = answer
-			else 
+			else
 				puts _("An example filename is now:\n\n\t%s") % [get_example_filename_various(@settings['basedir'], answer)]
 				@settings['naming_various'] = answer
 			end
@@ -211,27 +213,27 @@
 	
 	def get_answer(question, answer, default)
 		succes = false
-		while !succes 
+		while !succes
 			if answer == "yes"
 				answers = [_("yes"), _("y"), _("no"), _("n")]
-				STDOUT.print(question + " [#{default}] ")
+				STDOUT.print(question + " [#{default}] "); STDOUT.flush
 				input = STDIN.gets.strip
 				if input == '' ; input = default end
 				if answers.include?(input)
 					if input == _('y') || input == _("yes") ; return true else return false end
 				else puts _("Please answer yes or no") end
 			elsif answer == "open"
-				STDOUT.print(question + " [#{default}] ")
+				STDOUT.print(question + " [#{default}] "); STDOUT.flush
 				input = STDIN.gets.strip
 				if input == '' ; return default else return input end
 			elsif answer == "number"
-				STDOUT.print(question + " [#{default}] ")
+				STDOUT.print(question + " [#{default}] "); STDOUT.flush
 				input = STDIN.gets.strip
 				#convert the answer to an integer value, if input is a text it will be 0.
 				#make sure that the valid answer of 0 is respected though
 				if input.to_i == 0 && input != "0" ; return default else return input.to_i end
 			else puts _("We should never get here") ; puts _("answer = %s, question = %s (error)") % [answer, question]
- 			end
+			end
 		end
 	end
 
@@ -245,7 +247,7 @@
 			else
 				showFreedb()
 			end
-		else 
+		else
 			puts @settings['cd'].error
 			edit_settings()
 		end
