diff --git a/configure.ac b/configure.ac
index 1fcf149..dc6a4ad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -80,7 +80,7 @@ AC_CONFIG_SUBDIRS([gtk/phodav])
 SPICE_GTK_MAJOR_VERSION=`echo $PACKAGE_VERSION | cut -d. -f1`
 SPICE_GTK_MINOR_VERSION=`echo $PACKAGE_VERSION | cut -d. -f2`
 SPICE_GTK_MICRO_VERSION=`echo $PACKAGE_VERSION | cut -d. -f3 | cut -d- -f1`
-AS_IF([test "x$SPICE_GTK_MICRO_VERSION" = "x"], [SPICE_GTK_MICRO_VERSION = 0])
+AS_IF([test "x$SPICE_GTK_MICRO_VERSION" = "x"], [SPICE_GTK_MICRO_VERSION=0])
 AC_SUBST(SPICE_GTK_MAJOR_VERSION)
 AC_SUBST(SPICE_GTK_MINOR_VERSION)

@@ -270,6 +270,9 @@ AC_SUBST(GTHREAD_CFLAGS)
 AC_SUBST(GTHREAD_LIBS)
 PKG_CHECK_MODULES(SOUP, libsoup-2.4)
+AC_SUBST(SOUP_CFLAGS)
+AC_SUBST(SOUP_LIBS)
+
 AC_ARG_WITH([audio],
 AS_HELP_STRING([--with-audio=@<:@gstreamer/pulse/auto/no@:>@], [Select audio backend @<:@default=auto@:>@]),
