--- Makefile.in	2014-02-10 15:51:40.000000000 +0200
+++ Makefile.in	2014-02-18 03:06:56.000000000 +0200
@@ -479,7 +479,7 @@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
 NULL = 
-SUBDIRS = spice-common gtk po doc data tests $(am__append_1)
+SUBDIRS = spice-common gtk po doc data $(am__append_1)
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = spice-client-glib-2.0.pc \
    spice-client-gtk-$(SPICE_GTK_API_VERSION).pc $(NULL) \
